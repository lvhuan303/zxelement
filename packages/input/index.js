import ZxInput from './src/input.vue'

ZxInput.install = Vue => {
  Vue.component(ZxInput.name, ZxInput)
}

export default ZxInput
