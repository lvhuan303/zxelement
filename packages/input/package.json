{
  "name": "zx-input",
  "version": "0.0.2",
  "description": "",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "del": "rimraf dist"
  },
  "keywords": [],
  "author": "教瘦 <54337467@qq.com>",
  "license": "MIT",
  "dependencies": {
    "lodash": "3"
  },
  "gitHead": "8e97c46919eb2d3bd72d6e4e387481779fe45747"
}
