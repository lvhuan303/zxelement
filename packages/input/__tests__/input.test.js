import input from '../src/input.vue'
import { mount } from '@vue/test-utils'

describe('zx-input', () => {

  // 测试函数
  test('input-text', () => {
    const wrapper = mount(input)
    console.log(wrapper.html())
    expect(wrapper.html()).toContain('input type="text"')
  })

  test('input-password', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password'
      }
    })
    console.log(wrapper.html())
    expect(wrapper.html()).toContain('input type="password"')
  })

  test('input-password', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
        value: 'admin'
      }
    })
    expect(wrapper.props('value')).toBe('admin')
  })

  // 设置快照
  test('input-snapshot', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'text',
        value: 'admin'
      }
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
