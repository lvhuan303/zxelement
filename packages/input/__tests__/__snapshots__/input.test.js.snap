// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zx-input input-snapshot 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;

exports[`zx-input input-snapshot1 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
